src/utils/sortIdentities.ts

```diff
@@ -0,0 +1,26 @@
+import { ProviderUser } from "src/@types/ProviderUser";
+
+import { IdentityTypes, Identity } from "../@types/Identity";
+import { SortedIdentities } from "../@types/SortedIdentities";
+
+export function sortIdentities(fetchedData: {
+  data?: { provider: ProviderUser } | null;
+}): SortedIdentities {
+  if (fetchedData.data) {
+    const identities: Identity[] = fetchedData.data.provider.user.identities;
+    const phoneIdentities: Identity[] = [];
+    const emailIdentities: Identity[] = [];
+
+    identities.forEach((identity) => {
+      if (identity.type === IdentityTypes.EMAIL.toLowerCase()) {
+        emailIdentities.push(identity);
+      } else if (identity.type === IdentityTypes.PHONE.toLowerCase()) {
+        phoneIdentities.push(identity);
+      }
+    });
+
+    return { phoneIdentities, emailIdentities };
+  }
+
+  return { phoneIdentities: [], emailIdentities: [] };
+}
```

Comment:
```suggestion
export function sortIdentities(fetchedData: {
  data?: { provider: ProviderUser } | null;
}): SortedIdentities {
  if (fetchedData.data) {
    const identities: Identity[] = fetchedData.data.provider.user.identities;
    const phoneIdentities: Identity[] = [];
    const emailIdentities: Identity[] = [];

    identities.forEach((identity) => {
      if (identity.type === IdentityTypes.EMAIL.toLowerCase()) {
        emailIdentities.push(identity);
      } else if (identity.type === IdentityTypes.PHONE.toLowerCase()) {
        phoneIdentities.push(identity);
      }
    });
  }

  return { phoneIdentities, emailIdentities };
}
```

You can go with this, as `phoneIdentities` and `emailIdentities` are initialized with an empty array. If there are, for example, no phone identities, `phoneIdentities` won't be populated and will return an empty array 😁.

Reply:
OK fixed ee7d9ff